publication-rest/src/main/java/no/unit/nva/publication/services/model/UriMap.java

```diff
@@ -0,0 +1,32 @@
+package no.unit.nva.publication.services.model;
+
+import static java.util.Objects.isNull;
+import java.net.URI;
+import java.time.Instant;
+import no.unit.nva.identifiers.SortableIdentifier;
+import nva.commons.core.StringUtils;
+import nva.commons.core.paths.UriWrapper;
+
+public record UriMap(URI shortenedUri, URI longUri, Instant createdDate, Long expiresDate) {
+
+    private static final String SHORTENED_PATH = "file";
+
+    public static UriMap create(URI longVersion, Instant expiresDate, URI domain, String basePath) {
+        validateRequest(longVersion, expiresDate);
+        return new UriMap(createNewShortVersion(domain, basePath), longVersion, Instant.now(), expiresDate.getEpochSecond());
+    }
+
+    private static void validateRequest(URI longVersion, Instant expiresDate) {
+        if (isNull(longVersion) || StringUtils.isBlank(longVersion.toString()) || isNull(expiresDate)) {
+            throw new IllegalArgumentException("Missing required parameters");
```

Comment:
StringUtils.isBlank(longVersion.toString()) does null check by default, not sure it is necessary to have own null check for it.